### PR TITLE
Script for bumping services

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: scripts/bump.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            + "'"
+        )
+    latest_tag_data, *_ = response.json()
+    _repo_version_cache[repo] = latest_tag_data["name"]
+    return _repo_version_cache[repo]
+
+
+def get_open_aea_dependencies() -> t.Dict[str, str]:
+    """Get open-aea dependencies."""
+    version = get_latest_tag(OPEN_AEA_REPO).replace("v", "")
```
Comment (DavidMinarsch) [inline scripts/bump.py:47]: Works most of the time. But what if they're different...? 

More robust approach: scan the repo to be bumped for `open-aea-*` deps and then search for the correct new version in OPEN_AEA_REPO

File: scripts/bump.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        "open-aea-ledger-cosmos": f"=={version}",
+        "open-aea-ledger-solana": f"=={version}",
+        "open-aea-cli-ipfs": f"=={version}",
+    }
+
+
+def get_open_autonomy_dependencies() -> t.Dict[str, str]:
+    """Get open-autonomy dependencies."""
+    version = get_latest_tag(OPEN_AUTONOMY_REPO).replace("v", "")
+    return {
```
Comment (DavidMinarsch) [inline scripts/bump.py:62]: Same as above

File: scripts/bump.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
+"""Script for bumping core dependencies."""
```
Comment (DavidMinarsch) [inline scripts/bump.py:1]: Provide a short docstring explaining which actions are being taken

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: docs/guides/bumping_services.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+To bump a service to the latest version of open-autonomy follow these steps
+
+1. Bump open-autonomy and relevant dependencies to the desired version in `tox.ini`, `pipfile/pyproject.toml` and `packages`. You can also use [this](https://github.com/valory-xyz/open-autonomy/blob/main/scripts/bump.py) script to bump the dependencies in your repository.
```
Comment (angrybayblade) [inline docs/guides/bumping_services.md:3]: The CI will fail because of this until this PR is merged into main

File: scripts/bump.py [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+    if sync:
+        pm = PackageManagerV1.from_dir(
+            Path.cwd() / PACKAGES, config_loader=load_configuration
+        )
+        pm.sync(
+            sources=[
+                f"{OPEN_AEA_REPO}:{_version_cache[OPEN_AEA_REPO]}",
+                f"{OPEN_AUTONOMY_REPO}:{_version_cache[OPEN_AUTONOMY_REPO]}",
+            ]
+        )
```
Comment (Adamantios) [inline scripts/bump.py:275]: What if we need more sources? For example `contribution` needs IEKit as a source when syncing cc @dvilelaf.

**Review by Adamantios — COMMENTED** [review]
LGTM, left a comment.

File: scripts/bump.py (hunk 5 above)
Comment (dvilelaf) [reply scripts/bump.py]: Yes, it would be nice if we can specify those

**Review by Adamantios — APPROVED** [review]

**Review by dvilelaf — COMMENTED** [review]
LGTM. I agree with Manos. 

Has this been tested again one of our services?

**Review by dvilelaf — APPROVED** [review]

**Pull request merged**